### PR TITLE
fix: learner tab is hidden from learners

### PR DESCRIPTION
**Pull request by mehaknasir**
- learner tab will not be visible to student roles 
- learners won't be able to access it through URL as well

### CODE REVIEW

File: src/discussions/data/hooks.js [hunk 1]
```diff
@@ -167,3 +167,13 @@ export const useAlertBannerVisible = (content) => {
     || (content.abuseFlagged && canSeeReportedBanner)
   );
 };
+
+export const useIsShowLearnersTab = () => {
```
Comment (asadazam93) [inline src/discussions/data/hooks.js:171]: nit: useShowLearnersTab

File: src/discussions/data/hooks.js [hunk 2]
```diff
@@ -167,3 +167,13 @@ export const useAlertBannerVisible = (content) => {
...
+  const learnersTabEnabled = useSelector(selectLearnersTabEnabled);
+  const userRoles = useSelector(selectUserRoles);
+  const isAdmin = useSelector(selectUserIsStaff);
+  const isGTA = useSelector(selectUserIsGroupTa);
```
Comment (asadazam93) [inline src/discussions/data/hooks.js:175]: IsGroupTA

File: src/discussions/data/hooks.js [hunk 3]
```diff
@@ -167,3 +167,13 @@ export const useAlertBannerVisible = (content) => {
...
+  const privileged = useSelector(selectUserHasModerationPrivileges);
+  const allowedUsers = isAdmin || isGTA || privileged || (userRoles.includes('Student') && userRoles.length > 1);
```
Comment (asadazam93) [inline src/discussions/data/hooks.js:177]: Do we need the the first 3 checks here? If the userRoles includes Student and there is more than 1 role then the additional role will cover the first 3 checks here?

**Review by asadazam93 — APPROVED** [review]

**Review by awais-ansari — CHANGES_REQUESTED** [review]
we also need to add a condition on the post summary card username redirection. Click on a post summary card username redirects a user to the learner area. So we need to disable the redirection link for learners and should show a plan text username.

File: src/discussions/common/AuthorLabel.jsx [hunk 4]
```diff
@@ -41,6 +42,10 @@ function AuthorLabel({
 
   const className = classNames('d-flex align-items-center', labelColor);
 
+  const showUserNameAsLink = useShowLearnersTab()
+      && linkToProfile && author
+      && learnersTabEnabled && author !== messages.anonymous;
+
```
Comment (awais-ansari) [inline src/discussions/common/AuthorLabel.jsx:48]: You already checking the `learnersTabEnabled` in `useShowLearnersTab()`. So you can remove it here. 
nit: 
`const showUserNameAsLink = useShowLearnersTab() && linkToProfile && author 
&& author !== messages.anonymous;`

**Review by awais-ansari — APPROVED** [review]

File: src/discussions/common/AuthorLabel.jsx (hunk 4 above)
Comment (mehaknasir) [reply src/discussions/common/AuthorLabel.jsx]: done

**Pull request merged**